addons/survey/static/src/question_page/question_page_one2many_field.js

```diff
@@ -12,6 +15,92 @@ class QuestionPageOneToManyField extends X2ManyField {
+        // immediately during question creation.
+        // Preparing everything in order to override `this._openRecord` below.
+        const { saveRecord: superSaveRecord, updateRecord: superUpdateRecord } = useX2ManyCrud(
+            () => this.list,
+            this.isMany2Many
+        );
+
+        const self = this;
+        const saveRecord = async (record) => {
+            const saveResponse = await superSaveRecord(record);
+            try {
+                await self.props.record.save({stayInEdition: true, throwOnError: true});
+            } catch (error) {
+                return self.handleSurveySaveError(error, record);
+            }
+            return saveResponse;
+        };
+
+        const updateRecord = async (record) => {
+            const updateResponse = await superUpdateRecord(record);
+            try {
+                await self.props.record.save({stayInEdition: true, throwOnError: true});
+            } catch (error) {
+                return self.handleSurveySaveError(error);
+            }
+            return updateResponse;
+        };
+
+        const openRecord = useOpenX2ManyRecord({
+            resModel: this.list.resModel,
+            activeField: this.activeField,
+            activeActions: this.activeActions,
+            getList: () => this.list,
+            saveRecord,
+            updateRecord,
+        });
+        this._openRecord = async (params) => {
+            if (!await self.props.record.save({stayInEdition: true})) {
+                return;
+            }
+            const activeElement = document.activeElement;
+            if (params.record) {
+                // Trigger loading for values that may have changed when reordering/deleting
+                // rows (allowed_triggering_question_ids, is_conditional, is_placed_before_trigger).
+                await params.record.load();
+            }
+            openRecord({
+                ...params,
+                onClose: () => {
+                    if (activeElement) {
+                        activeElement.focus();
+                    }
+                },
+            });
+        };
+    }
+
+    /**
+     * As it is convenient to keep the questions modal open, two things should
+     * be cared for in case of error while saving the form after adding or
+     * updating a question:
+     *   * Remove erroneous question row added
+     *   * Replace default error modal with a notification
+     *
+     *   @param {Error} error Error thrown when saving survey/question.
+     *   @param {Record} recordToDelete (optional) In case the error is
+     *   thrown when saving a new question, it should be deleted from the
+     *   list.
+     */
+    async handleSurveySaveError(error, recordToDelete) {
+        error.event.preventDefault();
+        if (recordToDelete) {
+            const listRecord = this.list.records.find(r => r.__bm_handle__ === recordToDelete.__bm_handle__);
```

Comment:
Can't we identify them in another way? Using some "id" property?

I'm not a fan of double-underscore stuff as it generally means "don't use it, internal api only", unless we don't have a choice.